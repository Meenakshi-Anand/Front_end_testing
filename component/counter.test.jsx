import Counter from './counter';
import {shallow}  from 'enzyme';
import React, {Component} from 'react';
import Foo from './foo';
import renderer from 'react-test-renderer';
describe('Counter',()=>{
  it("starts with the count of 0",() => {
    let container = shallow(<Counter />);
    let counterCount = container.state().count;
    expect(counterCount).toBe(0);
  });
});
it("renders 'count' foos",() => {
  let container = shallow(<Counter />);

  expect(container.find("Foo").length).toBe(0);

  container.find("button").simulate("click");
  expect(container.find("Foo").length).toBe(1);

  container.find("button").simulate("click");
  container.find("button").simulate("click");
  container.find("button").simulate("click");
  expect(container.find("Foo").length).toBe(4);

});

it("matches the initial snapshot",()=>{
  const component = renderer.create(<Counter />);
  expect(component.toJSON()).toMatchSnapshot();
});
