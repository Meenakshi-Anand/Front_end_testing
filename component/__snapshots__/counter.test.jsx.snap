// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the initial snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  />
</div>
`;
